google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1beta2/stub/EnhancedBigQueryReadStub.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigquery.storage.v1beta2.stub;
+
+import com.google.api.core.InternalApi;
+import com.google.api.gax.core.BackgroundResource;
+import com.google.api.gax.rpc.ClientContext;
+import com.google.api.gax.rpc.ServerStreamingCallable;
+import com.google.api.gax.rpc.UnaryCallable;
+import com.google.cloud.bigquery.storage.v1beta2.CreateReadSessionRequest;
+import com.google.cloud.bigquery.storage.v1beta2.ReadRowsRequest;
+import com.google.cloud.bigquery.storage.v1beta2.ReadRowsResponse;
+import com.google.cloud.bigquery.storage.v1beta2.ReadSession;
+import com.google.cloud.bigquery.storage.v1beta2.SplitReadStreamRequest;
+import com.google.cloud.bigquery.storage.v1beta2.SplitReadStreamResponse;
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Enhanced stub class for BigQuery Storage API.
+ *
+ * <p>This class is for advanced usage and reflects the underlying API directly.
+ */
+public class EnhancedBigQueryReadStub implements BackgroundResource {
+  private final GrpcBigQueryReadStub stub;
+
+  public static EnhancedBigQueryReadStub create(EnhancedBigQueryReadStubSettings settings)
+      throws IOException {
+    // Configure the base settings.
+    BigQueryReadStubSettings.Builder baseSettingsBuilder =
+        BigQueryReadStubSettings.newBuilder()
+            .setTransportChannelProvider(settings.getTransportChannelProvider())
+            .setEndpoint(settings.getEndpoint())
+            .setHeaderProvider(settings.getHeaderProvider())
+            .setCredentialsProvider(settings.getCredentialsProvider())
+            .setStreamWatchdogCheckInterval(settings.getStreamWatchdogCheckInterval())
+            .setStreamWatchdogProvider(settings.getStreamWatchdogProvider());
+
+    // CreateReadSession is a simple pass-through.
+    baseSettingsBuilder
+        .createReadSessionSettings()
+        .setRetryableCodes(settings.createReadSessionSettings().getRetryableCodes())
+        .setRetrySettings(settings.createReadSessionSettings().getRetrySettings());
+
+    // ReadRows is a simple pass-through.
```

Comment:
what is meant by pass through doesn setting the resumption strategy really make this case.

Reply:
The pass-through comment is a leftover from v1beta1, when we had more customization.

Removed the comment to avoid confusion in a separate commit: https://github.com/googleapis/java-bigquerystorage/pull/48/commits/5675ea27c43881739d51a6dd490f666d372e437f